pkg/transaction/metric/cachestorage.go

```diff
@@ -147,7 +147,7 @@ func (c *cacheStorage) loadAPIMetric(storageCache cache.Cache) {
 
 			var apiStatusMetric *APIMetric
 			for _, duration := range apiMetric.Values {
-				apiStatusMetric = c.collector.updateMetric(APIDetails{apiMetric.API.ID, apiMetric.API.Name, apiMetric.API.Revision}, apiMetric.StatusCode, duration)
+				apiStatusMetric = c.collector.updateMetric(APIDetails{apiMetric.API.ID, apiMetric.API.Name, apiMetric.API.Revision}, apiMetric.StatusCode, duration, apiMetric.Team)
```

Comment:
since this api will always be part of the same team, maybe put the team details in API Details object, as to not change the signature of this method

Reply:
well what if the API ownership changes? I know that its not possible in v7, but not sure about other gateways

Reply:
but it can be part of the APIDetails object that we are creating here, should not matter if it changes.

Reply:
yeah you're right, I realized it later and deleted my comment but then it was too late :)

Reply:
`apiStatusMetric = c.collector.updateMetric(APIDetails{apiMetric.API.ID, apiMetric.API.Name, apiMetric.API.Revision, apiMetric.Team}, apiMetric.StatusCode, duration`

`type APIDetails struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	Revision int    `json:"revision"`
        TeamID string `json:"team"`
}`

Reply:
Team ID can be extracted from API object 
`apiStatusMetric = c.collector.updateMetric(APIDetails{apiMetric.API.ID, apiMetric.API.Name, apiMetric.API.Revision, apiMetric.API.TeamID}, apiMetric.StatusCode, duration)`